### ISSUE TITLE
D1 queries like `select ?1 + ?2` broken after #480 PR

### ISSUE THREAD

**Issue by ruslantalpa** [cloudflare/miniflare#526]
I added these lines to one of the tests (to make sure it's not my code):
```
const sum = await db.prepare("SELECT ?1 + ?2 as sum").bind(1, 2).first("sum");
t.is(sum, 3);
```

and the test fails
```
d1 › test › d1js.ts › D1PreparedStatement: first

  packages/d1/test/d1js.spec.ts:212

   211:
   212:   const sum = await db.prepare("SELECT ?1 + ?2 as sum").bind(1, 2).first("sum");
   213:   t.is(sum, 3);

  Rejected promise returned by test. Reason:

  Error {
    message: 'D1_ERROR',
  }

  › D1Database._send (packages/d1/src/d1js.ts:148:13)
  › D1PreparedStatement.first (packages/d1/src/d1js.ts:215:7)
  › packages/d1/test/d1js.spec.ts:212:15

  ─
```

`statements.ts` used to have this code which was deleted
```
      // For statements using ?1 ?2, etc, we want to pass them as varargs but
      // "better" sqlite3 wants them as an object of {1: params[0], 2: params[1], ...}
      if (this.#bindings.length > 0 && typeof this.#bindings[0] !== "object") {
        return prepared.bind(
          Object.fromEntries(this.#bindings.map((v, i) => [i + 1, v]))
        );
      } else {
        throw e;
      }
```

**Comment by mrbbot** [comment]
Hey! 👋 Thanks opening this issue. It looks like a duplicate of https://github.com/cloudflare/miniflare/issues/526, so I'm going to close this and keep tracking everything over there. 👍
